### ISSUE TITLE
Narratives cannot display frequencies

### ISSUE THREAD

**Issue by jameshadfield** [nextstrain/auspice#1283]
Narratives cannot display frequencies panels in Auspice 2.33.0. I believe this _used_ to work in a previous version, but have not bisected to confirm. A test narrative showing this has been created on the [bug-narratives-frequencies](https://github.com/nextstrain/auspice/tree/bug-narratives-frequencies) branch -- please base any fix on that branch!

**How to reproduce**  

Steps to reproduce the current behavior:  
1. `get fetch`
2. `git checkout bug-narratives-frequencies`
3. `npm run get-data`
4. `auspice develop --narrativeDir narratives`
5. load http://localhost:4000/narratives/test/display-frequencies-panel

**Additional context**  

The frequencies sidecar JSON is not fetched when the narrative is loaded. When loading http://localhost:4000/ncov/global?d=tree,frequencies, the dataset the narrative is attempting to display and which correctly displays the frequencies when viewed by itself, the following `getDataset` requests are made:
```
http://localhost:4000/charon/getDataset?prefix=/ncov/global
http://localhost:4000/charon/getDataset?prefix=/ncov/global&type=tip-frequencies
http://localhost:4000/charon/getDataset?prefix=/ncov/global&type=root-sequence
```

However when loading the narrative, only the first of these is made. Auspice should be making all 3 requests in narratives mode.

P.S. I highly suspect that narratives won't be able to colour by non-varying bases, as the `root-sequence` JSON is not fetched either! I haven't created a narrative to test this, but will do at some point.

**Comment by jameshadfield** [comment]
I've looked into this and the 🐛 arrived when we allowed multiple different datasets per narrative. When a narrative arrives, we call the function `fetchAndCacheNarrativeDatasets`, which (a) fetches the main dataset JSON and (b) fetches other dataset JSONs defined in the narrative and stores them in redux via `CACHE_JSONS`. This contrasts with loading a non-narrative page, where we use the function `fetchDataAndDispatch`, which not only fetches the main JSON but also sidecar JSONs, as required. 

Any solution should unify the dataset-fetching code into a single function (either async fn or async generator), which both narratives and non-narratives use. Additionally, the cache structure, currently simply `jsons[treeName]=jsonData`, will need to be expanded to `jsons[treeName] = {main: jsonData, "tip-frequencies": jsonData, ...}`.
